core/src/main/java/io/grpc/internal/ServiceConfigFile.java

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright 2017, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a ServiceConfig as described at
+ * https://github.com/grpc/grpc/blob/master/doc/service_config.md
+ */
+public final class ServiceConfigFile {
```

Comment:
Please document that all fields are nullable.

Reply:
Done.   All fields are nullable so that this class can represent the json appropriately.   That said, some of the nullable fields are actually required